frontend/test/resolution/testResolve.cpp

```diff
@@ -723,6 +723,113 @@ static void test16() {
   }
 }
 
+// module-level split-init variables
+static void test17() {
+  Context context;
+
+  auto variables = resolveTypesOfVariables(&context,
+      R"""(
+      var foo;
+      foo = 5;
+
+      proc setArgToStr(out arg: string) {
+        arg = "str";
+      }
+      var bar;
+      setArgToStr(bar);
+
+      param foo_param;
+      foo_param = 5;
+      param bar_param;
+      bar_param = "bar_param";
+
+      )""", { "foo", "bar", "foo_param", "bar_param"});
+
+  auto foo = variables.at("foo");
+  assert(foo.kind() == QualifiedType::VAR);
+  assert(foo.type());
+  assert(foo.type()->isIntType());
+
+  auto bar = variables.at("bar");
+  assert(bar.kind() == QualifiedType::VAR);
+  assert(bar.type());
+  assert(bar.type()->isStringType());
+
+  ensureParamInt(variables.at("foo_param"), 5);
+  ensureParamString(variables.at("bar_param"), "bar_param");
+}
+
+// invalid module-level split-init
+static void test18() {
+  Context context;
+  // Make sure no errors make it to the user, even though we will get errors.
+  ErrorGuard guard(&context);
+
+  auto variables = resolveTypesOfVariables(&context,
+      R"""(
+      var flag = true;
+      var foo;
+      if (flag) {
+        foo = 5;
+      } else {
+        foo = "asdf";
+      }
+      )""", {"foo"});
+
+  assert(guard.numErrors() == 1);
+  assert(guard.error(0)->message() ==
+         "types do not match in conditional split init");
```

Comment:
That does sound good -- I'll do it in this PR

Reply:
Done, but it turns out we do not keep track of initialization-point IDs in `VarFrame` so I have used conditional blocks instead of initialization parts to position the notes in the error, with a TODO.